test/src/tests-serverModel.ts

```diff
@@ -0,0 +1,1675 @@
+                Timestamp: 0,
+                EventDataType: 0,
+                Debug: true,
+                CurrencyCode: 'USD',
+            };
+
+            ServerModel.appendUserInfo(user, event as SDKEvent);
+
+            expect(event.MPID).to.equal('98765');
+            expect(event.ConsentState).to.eql({
+                gdpr: {
+                    'test-purpose': {
+                        consented: true,
+                        timestamp: 11111,
+                        consent_document: 'gdpr-doc',
+                        location: 'test-gdpr-location',
+                        hardwareId: 'test-gdpr-hardware',
+                    },
+                },
+                ccpa: {
+                    data_sale_opt_out: {
+                        consented: false,
+                        timestamp: 22222,
+                        consent_document: 'ccpa-doc',
+                        location: 'test-ccpa-location',
+                        hardwareId: 'test-ccpa-hardware',
+                    },
+                },
+            });
+            expect(event.UserAttributes).to.eql({
+                'foo-user-attr': 'foo-attr-value',
+                'foo-user-attr-list': ['item1', 'item2'],
+            });
+            expect(event.UserIdentities).to.eql([
+                { Identity: 'foo-other', Type: 0 },
+                { Identity: '1234567', Type: 1 },
+                { Identity: 'foo-email', Type: 7 },
+                { Identity: 'foo-other2', Type: 10 },
+                { Identity: 'foo-other3', Type: 11 },
+                { Identity: 'foo-other4', Type: 12 },
+            ]);
+        });
+
+        it('sets User Attributes and Identities to null if user is empty', () => {
+            const event = {} as SDKEvent;
+
+            ServerModel.appendUserInfo(null, event);
+            expect(event.MPID).to.eql(null);
+            expect(event.ConsentState).to.eql(null);
+            expect(event.UserIdentities).to.eql(null);
+            expect(event.UserAttributes).to.eql(null);
+        });
+
+        it('returns undefined if event is empty', () => {
+            expect(ServerModel.appendUserInfo({} as MParticleUser, null)).to.be
+                .undefined;
+        });
+
+        it('returns early if event MPID matches User MPID', () => {
+            const user = {
+                getUserIdentities: () => {
+                    return { userIdentities: {} };
+                },
+                getAllUserAttributes: () => {
+                    return null;
+                },
+                getMPID: () => {
+                    return '123456';
+                },
+                getConsentState: () => {
+                    return null;
+                },
+            };
+
+            const event = {
+                MPID: '123456',
+            };
+
+            ServerModel.appendUserInfo(user, event as SDKEvent);
+
+            expect(event.MPID).to.equal('123456');
```

Comment:
Can we add an expect for event to only have the key of `MPID` and nothing else?  Something like
```suggestion
            expect(event.MPID).to.equal('123456');
            expect(Object.keys(event).length.to.equal(1)
```